admin/src/pages/SettingsPage/index.tsx

```diff
@@ -167,16 +195,35 @@ const SettingsPage = () => {
       };
     }
     return ct;
-  });
+  }) : [];
   const selectedContentTypes = configContentTypes.map(item => item.uid);
-  const audienceFieldChecked = navigationConfigData?.additionalFields.includes(navigationItemAdditionalFields.AUDIENCE);
+  const audienceFieldChecked = navigationConfigData?.additionalFields?.includes(navigationItemAdditionalFields.AUDIENCE);
   const allowedLevels = navigationConfigData?.allowedLevels || 2;
   const nameFields = navigationConfigData?.contentTypesNameFields || {}
   const populate = navigationConfigData?.contentTypesPopulate || {}
   const i18nEnabled = navigationConfigData?.i18nEnabled ?? false
   const isI18NPluginEnabled = navigationConfigData?.isI18NPluginEnabled;
   const defaultLocale = navigationConfigData?.defaultLocale;
 
+  const handleSubmitCustomField = (field: NavigationItemCustomField) => {
+    const filteredFields = customFields.filter(f => f.name !== field.name);
+    setCustomFields([...filteredFields, field]);
```

Comment:
```suggestion
    const fields = customFields.map(current => current.name === field.name ? field : current);
    setCustomFields(fields);
```

Reply:
At first, it looked like a great way to make this part cleaner. But after testing I realized why I hadn't done it this way in the first place. Solution with filter works for both situations when the name exists in `customFields` and when it does not. To make it simpler I could divide it into two separate methods:`handleNewCustomField` and `handleEditCustomField`.